### PR TITLE
B 11 get one vinyl with offers

### PR DESCRIPTION
**Pull request by Greengrer**

### CODE REVIEW

File: src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcVinylDao.java [hunk 1]
```diff
@@ -26,7 +26,9 @@
     private final String SELECT_MANY_RANDOM_UNIQUE_VINYLS = "SELECT id, release, artist, full_name, link_to_image " +
             "FROM unique_vinyls TABLESAMPLE SYSTEM_ROWS(?) LIMIT ?;";
     private final String SELECT_MANY_UNIQUE_VINYLS_BY_FULL_NAME_MATCH = "SELECT id, release, artist, full_name, link_to_image " +
-            "FROM unique_vinyls WHERE full_name LIKE ?";
+            "FROM unique_vinyls WHERE full_name ILIKE ?";
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcVinylDao.java:28]: Please make it case insensitive

File: src/main/java/com/vinylteam/vinyl/entity/Shop.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package com.vinylteam.vinyl.entity;
+
+import java.util.Objects;
+
+public class Shop {
+
+    int id;
+    String mainPageLink;
+    String imageLink;
+    String name;
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/entity/Shop.java]: Please add access modifiers

File: src/test/java/com/vinylteam/vinyl/web/servlets/OneVinylOffersServletTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        HttpServletRequest mockedRequest = mock(HttpServletRequest.class);
+        HttpServletResponse mockedResponse = mock(HttpServletResponse.class);
+        VinylService mockedVinylService = mock(DefaultVinylService.class);
+        ShopService mockedShopService = mock(DefaultShopService.class);
+        Vinyl mockedUniqueVinyl = mock(Vinyl.class);
+        Vinyl mockedVinyl = mock(Vinyl.class);
+        Shop mockedShop = mock(Shop.class);
+        List<Vinyl> vinylOffers = new ArrayList<>(List.of(mockedVinyl));
+        List<Integer> shopsIds = new ArrayList<>();
+        List<Shop> shopsByIds = new ArrayList<>(List.of(mockedShop));
```
Comment (AlexKievDeveloper) [inline src/test/java/com/vinylteam/vinyl/web/servlets/OneVinylOffersServletTest.java]: All this also applies to prepare

File: src/test/java/com/vinylteam/vinyl/service/impl/DefaultVinylServiceTest.java [hunk 4]
```diff
@@ -28,35 +26,21 @@
     private final Vinyl vinyl = new Vinyl();
     private final Vinyl uniqueVinyl = new Vinyl();
     private final List<Vinyl> uniqueVinyls = new ArrayList<>(List.of(uniqueVinyl));
+    private final List<Vinyl> vinyls = new ArrayList<>(List.of(vinyl));
```
Comment (AlexKievDeveloper) [inline src/test/java/com/vinylteam/vinyl/service/impl/DefaultVinylServiceTest.java]: private final List<Vinyl> vinyls = List.of(vinyl);

**Review by AlexKievDeveloper — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcShopDao.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                logger.error("Error while getting list of shops with ids from list of ids from db {'ids':{}, 'shops':{}}",
+                        ids, shops, e);
+                throw new RuntimeException(e);
+            }
+        }
+        logger.debug("Resulting list of shops with ids from list of ids is {'shops':{}}", shops);
+        return shops;
+    }
+
+    String fillSelectManyByIdsStatement(List<Integer> ids) {
```
Comment (TarasOrlovskyi) [inline src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcShopDao.java:47]: i think that this method should be private

File: src/main/java/com/vinylteam/vinyl/entity/Shop.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
+package com.vinylteam.vinyl.entity;
+
+import java.util.Objects;
+
+public class Shop {
+
+    private int id;
+    private String mainPageLink;
+    private String imageLink;
```
Comment (TarasOrlovskyi) [inline src/main/java/com/vinylteam/vinyl/entity/Shop.java:9]: I suggest to rename to logoLink

File: src/main/resources/static/index.html [hunk 7]
```diff
@@ -1,126 +1,127 @@
...
+                    <ul class="header__menu-list">
+                        <li>
+                            <a href="#" class="header__menu-item">about</a>
+                        </li>
+                        <li>
+                            <a href="catalog" class="header__menu-item">catalog</a>
+                        </li>
+                        <li>
+                            <a href="index.html" class="header__logo">
+                                <img src="img/svg/wax_deals.svg" alt="wax deals logo">
```
Comment (TarasOrlovskyi) [inline src/main/resources/static/index.html:33]: Don't do so please as i have set up 2 indents in webStorm. This makes it easier to view the html code

**Review by TarasOrlovskyi — DISMISSED** [review]
I approve, but consider my wishes, please

File: src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcShopDao.java (hunk 5 above)
Comment (Greengrer) [reply src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcShopDao.java:47]: I need it package private in order to test.

File: src/main/resources/static/index.html (hunk 7 above)
Comment (Greengrer) [reply src/main/resources/static/index.html:33]: Sorry, I'll change it.

File: src/main/java/com/vinylteam/vinyl/entity/Shop.java (hunk 6 above)
Comment (Greengrer) [reply src/main/java/com/vinylteam/vinyl/entity/Shop.java:9]: It is called link_to_image in db. Gotta change it too to match - so maybe later.

**Review by TarasOrlovskyi — APPROVED** [review]

**Review by AlexKievDeveloper — APPROVED** [review]

**Pull request merged**